src/sentry/static/sentry/app/stores/releaseStore.tsx

```diff
@@ -0,0 +1,217 @@
+import Reflux from 'reflux';
+
+import ReleaseActions from 'app/actions/releaseActions';
+import {Deploy, Release} from 'app/types';
+
+type StoreRelease = Map<string, Release>;
+type StoreDeploys = Map<string, Array<Deploy>>;
+type StoreLoading = Map<string, boolean>;
+type StoreError = Map<string, Error>;
+
+type ReleaseStoreInterface = {
+  get(
+    projectSlug: string,
+    releaseVersion: string
+  ): {
+    release: Release | undefined;
+    releaseLoading: boolean | undefined;
+    releaseError: Error | undefined;
+    deploys: Array<Deploy> | undefined;
+    deploysLoading: boolean | undefined;
+    deploysError: Error | undefined;
+  };
+
+  state: {
+    orgSlug: string | undefined;
+    release: StoreRelease;
+    releaseLoading: StoreLoading;
+    releaseError: StoreError;
+    deploys: StoreDeploys;
+    deploysLoading: StoreLoading;
+    deploysError: StoreError;
+  };
+
+  loadRelease(orgSlug: string, projectSlug: string, releaseVersion: string): void;
+  loadReleaseSuccess(projectSlug: string, releaseVersion: string, data: Release): void;
+  loadReleaseError(projectSlug: string, releaseVersion: string, error: Error): void;
+  loadDeploys(orgSlug: string, projectSlug: string, releaseVersion: string): void;
+  loadDeploysSuccess(projectSlug: string, releaseVersion: string, data: Release): void;
+  loadDeploysError(projectSlug: string, releaseVersion: string, error: Error): void;
+};
+
+export const getReleaseStoreKey = (projectSlug: string, releaseVersion: string) =>
+  `${projectSlug}${releaseVersion}`;
+
+const ReleaseStoreConfig: Reflux.StoreDefinition & ReleaseStoreInterface = {
+  state: {
+    orgSlug: undefined,
+    release: new Map() as StoreRelease,
+    releaseLoading: new Map() as StoreLoading,
+    releaseError: new Map() as StoreError,
+    deploys: new Map() as StoreDeploys,
+    deploysLoading: new Map() as StoreLoading,
+    deploysError: new Map() as StoreError,
+  },
+
+  listenables: ReleaseActions,
```

Comment:
Should you also listen to the organization actions to reset state when organizations are changed?

Reply:
Great suggestion 😁